packages/mgt-components/src/components/preview/mgt-search-results/mgt-search-results.ts

```diff
@@ -525,7 +525,7 @@ export class MgtSearchResults extends MgtTemplatedComponent {
             const thumbnailBatch = graph.createBatch<BinaryThumbnail>();
             const thumbnailBatchBeta = BetaGraph.fromGraph(graph).createBatch<BinaryThumbnail>();
 
-            for (let i = 0; i < response.value[0].hitsContainers[0].hits.length; i++) {
+            for (let i = 0; i < response.value[0].hitsContainers[0].hits?.length; i++) {
```

Comment:
Could this be a for-of loop?
```suggestion
            for (const element of response.value[0].hitsContainers[0].hits ?? []) {
```

Reply:
It also strikes me... But it seems to be the way the API was implemented... We could maybe be more protective on using the arrays there? To ensure they are populated etc.?

`for (const element of response.value?.[0].hitsContainers?.[0].hits ?? []) {`

Reply:
the optional chaining operator will only check that the array exists, not that there are elements in the array.

Updated the suggestion with a safer access to the hits property